### PR TITLE
Fixes the arm64 image build issue

### PR DESCRIPTION
**Pull request by chenhao94**

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -229,10 +229,11 @@ RUN set -e; \
                 jq \
                 lib{asan6,lsan0,tsan0,ubsan1} \
                 $(! echo "$DISTRO_ID-$DISTRO_VERSION_ID" | grep -e'^ubuntu\-22\.04$' >/dev/null || echo lib{asan8,tsan2}) \
-                lib{benchmark,boost-all,curl4-openssl,edit,eigen3,event,gflags,gmp,gtest,hiredis,hwloc,leveldb,lmdb,lzma,mpfr,mpc,ncurses5,rocksdb,snappy,ssl,tbb}-dev \
+                lib{benchmark,boost-all,curl4-openssl,edit,eigen3,event,gflags,gmp,gtest,hdf5,hiredis,hwloc,leveldb,lmdb,lzma,mpfr,mpc,ncurses5,rocksdb,snappy,ssl,tbb}-dev \
```
Comment (xkszltl) [inline docker/Dockerfile:232]: Why? IIRC py lib should provide the matching version of hdf5 for use. Distro installation may not match the required version.

**Review by xkszltl — APPROVED** [review]

File: docker/Dockerfile (hunk 1 above)
Comment (chenhao94) [reply docker/Dockerfile:232]: In the arm build logs, you can see they are all complaining about missing libhdf5.so when installing h5py, starting from https://github.com/cyfitech/cris-core/actions/runs/8639437061. Maybe they did something weird recently.

Comment (chenhao94) [reply docker/Dockerfile:232]: Any ideas?

Comment (xkszltl) [reply docker/Dockerfile:232]: It's likely because they no longer build for arm. You'll need to dig in to figure out how they treat x86/arm differently.

Comment (chenhao94) [reply docker/Dockerfile:232]: Reopen to keep the discussion

**Pull request merged**